File: PDOO/Napakalaki/src/NapakalakiGame/Treasure.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

/**
 *
 * @author nestor
 */
public class Treasure {

    private String name;
    private int bonus;
    private TreasureKind type;

    public Treasure(String n, int bonus, TreasureKind T) {
        name = n;
        this.bonus = bonus;
        type = T;
    }

    public String getName() {
        return name;
    }

    public int getBonus() {
        return bonus;
    }

    public TreasureKind getType() {
        return type;
    }

    public String toString() {
        return "Name: " + name + ", Bonus: " + bonus + ", Type: " + type;
    }
}

File: PDOO/Napakalaki-Ruby/lib/treasure.rb
#encoding: UTF-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module NapakalakiGame

  require_relative "treasure_kind.rb"

  class Treasure

    def initialize(name, bonus,t)
      @name=name
      @bonus=bonus
      @type=t
    end
  
    def getName
      return @name
    end
  
    def getBonus
      return @bonus
    end
  
    def getType
      return @type
    end
    
    def to_s
      "Name: #{@name}, Bonus: #{@bonus}, Type: #{@type}"
    end
  
    #Borrar luego, lo dejo para que main sirva
    attr_reader:name
    attr_reader:bonus
    attr_reader:type
  end

end
